### ISSUE TITLE
List of all possible rules that can be disabled?

### ISSUE THREAD

**Issue by autonomousapps** [jeremymailen/kotlinter-gradle#109]
I tried migrating to 2.0.0, and got a LOT of these errors:
> Imports must be ordered in lexicographic order without any empty lines in-between

How can I disable that rule? I only see an example for no-wildcard-imports. Is there a list somewhere?

**Comment by autonomousapps** [comment]
FYI, I tried just accepting this new rule and formatting my code, but then compilation failed :man_shrugging: I assume it's related to some bug in ktlint with imports (I recall seeing some issue with `as` import aliasing), but I can't be assed to look it up right now. I'd prefer to just disable the rule and then re-enable it later, if ever.

**Comment by jeremymailen** [comment]
Yeah, there's definitely some changes in the new ktlint. `import-ordering` and `no-unused-imports` are two rule ids you may need. In the checkstyle report you can see the rule id of each failure as `source`:
```xml
<?xml version="1.0" encoding="utf-8"?>
<checkstyle version="8.0">
        <file name="src/main/kotlin/com/gladly/gateways/CommunicationQueueGateway.kt">
                <error line="3" column="1" severity="error" message="Imports must be ordered in lexicographic order without any empty lines in-between" source="import-ordering" />
...
```

**Comment by jeremymailen** [comment]
I would hesitate to try to keep a doc up-to-date with all the rule ids.

What if we include the rule id that failed in the console output as well? This will make it easier to debug issues.

The alternative would be a task to list all rule sets and ids -- might be valuable at some point.

**Comment by autonomousapps** [comment]
Thanks for that quick response. I would say that including the rule ID in the console output might be sufficient.

One problem I just discovered is that the new `disabledRules` property is only available on your extension DSL. Since I have a custom Lint task, that doesn't work. This workaround does, though:
```
def lintKotlinAll = tasks.register('lintKotlinAll', LintTask) {
    group = 'verification'
    source files(chessProjects.collect { "${it.basePath}/src" })
    ktLintParams = ktLintParams.copy(
        ktLintParams.indentSize,
        ktLintParams.continuationIndentSize,
        ktLintParams.experimentalRules,
        ["import-ordering", "final-newline", "no-unused-imports"] as String[],
        ktLintParams.editorConfigPath
    )
    reports = [
        'plain': file("$buildDir/reports/ktlint/all-lint.txt"),
        'html' : file("$buildDir/reports/ktlint/index.html")
    ]
}
```
Please note that I had to add `"no-unused-imports"` because of that issue with ktlint and `as` aliases. It was reporting imports like this as unused, even when they are:
```
import com.myapp.appbase.R as AppBaseR
```
I probably won't update to 2.0.0 due to this. Hopefully 2.0.1 comes soon :)

**Comment by jeremymailen** [comment]
It's good to know you are using that feature of custom tasks. I will add setters there that mirror the parameters available in the extension.

I'll also add the rule id to the console error output.

ktlint 0.34.1 just came out so I'll try to do an update tonight.

**Comment by jeremymailen** [comment]
Ok, just released all of the above in [2.1.0](https://github.com/jeremymailen/kotlinter-gradle/releases/tag/2.1.0). Let me know how it works out.

**Comment by autonomousapps** [comment]
It's perfect, thanks.
